crates/host/src/wasmbus/mod.rs

```diff
@@ -1431,4 +1606,151 @@ impl Lattice {
             _ => {}
         }
     }
+
+    #[instrument(skip(self, id, value))]
+    async fn process_linkdef_put(
+        &self,
+        id: impl AsRef<str>,
+        value: impl AsRef<[u8]>,
+    ) -> anyhow::Result<()> {
+        let id = id.as_ref();
+        let value = value.as_ref();
+        let ref ld @ LinkDefinition {
+            ref actor_id,
+            ref provider_id,
+            ref link_name,
+            ref contract_id,
+            ref values,
+            ..
+        } = serde_json::from_slice(value).context("failed to deserialize link definition")?;
+        ensure!(
+            id == linkdef_hash(actor_id, contract_id, link_name),
+            "linkdef hash mismatch"
+        );
+
+        debug!(
+            id,
+            actor_id,
+            provider_id,
+            link_name,
+            contract_id,
+            ?values,
+            "process link definition entry put"
+        );
+
+        let mut links = self.links.write().await;
+        links.insert(id.to_string(), ld.clone());
+
+        self.publish_event(
+            "linkdef_set",
+            event::linkdef_set(id, actor_id, provider_id, link_name, contract_id, values),
+        )
+        .await?;
+
+        // TODO: Broadcast `linkdef_added`
```

Comment:
I don't think this is needed. `process_linkdef_put` is called from `process_entry`, which is triggered by watching the KV bucket. Every host in the lattice should receive the same watch event and process it locally

This may have been needed in the OTP host, where links could be created through two different ways: direct writes to the bucket, or (legacy) messages on `wasmbus.ctl`. We won't need to support the latter here

(Note each host in the lattice will publish the `linkdef_set` event, resulting in duplicates. That's a known ugly behavior of wasmCloud hosts)